app/controllers/articles_controller.rb

```diff
@@ -27,6 +27,8 @@ def create
   def edit; end
 
   def update
+    # not saved if fail so we can always increment
+    @article.edition_counter += 1 
```

Comment:
E porque não um before_update no modelo? 
E como é que garantimos que se tivermos duas pessoas a editar ao mesmo tempo este número fica com o valor correcto?

Reply:
Podemos garantir isso recorrendo a, por exemplo, Optimistic Locking usando o edition counter para version lock
Assim, se o utilizador for tentar editar e o edition_counter já for diferente do dele, permite levantar o erro de StaleObject, onde podemos redirecionar o user de novo para a edição mas com os dados atualizados